### ISSUE TITLE
Feature request: provide webhooks

### ISSUE THREAD

**Issue by learntheropes** [Peach2Peach/peach-api-ts#1]
I see that you already provide a `fmcToken` property in the Update User endpoint.
I’m not sure what is that, but I guess is for mobile only, right? So I hope my suggestion is not a duplicate of that.

But to make a Peach web app, webhooks on new matches and new messages would be useful.

1. When the user post to the update user endpoint, it could optionally provide a destination URL for the webhooks.
2. On updates, the web app server gets a post request with encrypted data or no data at all from peach. From the URL where it's receiving, it knows which user to notify.
3. The server notifies the user by mail or telegram that there is an update on one of her/his trades.
4. The user opens the browser and fetches and decrypts the new data. 

PS: I am aware the browser is not the most secure option when it comes to bitcoin, but in my opinion is an acceptable tradeoff in some circumstances, like for a disposable account where a nocoiner is buying bitcoin to pay an invoice on Btcpay Server and then the mnemonic is destroyed.

**Comment by learntheropes** [comment]
PS: it may be beneficial also for the load of your server: instead of receiving a request every minute that checks for update (i.e. new message in the chat), you post just once when there is actually an update.

**Comment by Czino** [comment]
Yes, the `fcmToken` is used for sending out push notifications, whenever the server wants to notify the user. That's already happening when there is actually an update.

I like the webhook idea which could be provided on a user level that could be in addition to the `fcmToken`.  Ideally the webhook interface would be similar to the datastructure we already use to send out push notifications. 

We already use `NotificationType`s such as `contract.paymentMade`, `offer.sellOfferExpired`, `contract.chat`, etc. + an array of strings containing data like offer or contract id among other.

I'll leave this issue open, to come back to it, should it be implemented
